lib/request.js

```diff
@@ -433,12 +433,28 @@ module.exports = {
   get ips() {
     const proxy = this.app.proxy;
     const val = this.get(this.app.proxyIpHeader);
-    let ips = proxy && val
-      ? val.split(/\s*,\s*/)
-      : [];
+    let ips = [];
+
+    if (proxy && val) {
+      ips = val.split(/\s*,\s*/)
+        .map(host => {
+          let normalizedHost = host;
+
+          if (net.isIPv6(host)) {
+            normalizedHost = `[${host}]`;
+          }
+
+          const hostname = new URL(`http://${normalizedHost}`).hostname;
+
+          return hostname.replace(/(^\[|\]$)/g, '');
```

Comment:
@3imed-jaberi I don't think I can use `parseurl` module here as it expects a `http.IncomingMessage` object and here we're working on a header.  I see `url.parse` is now deprecated so I swapped to using the `URL` constructor.  

Problem is this doesn't strip the `[]` chars from IPv6 hosts so now we gotta do it manually.  Regex to the rescue